### PR TITLE
feat(FX-3266): Gracefully respond when Algolia API key expires

### PR DESCRIPTION
**Pull request by dimatretyak**
The type of this PR is: **Feature**

This PR resolves [FX-3266]

### Demo
https://user-images.githubusercontent.com/3513494/133488934-98c189e0-0ebf-4b00-bc9b-d2770a5bdbf5.mp4

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Gracefully respond when Algolia API key expires - dimatretyak

</details>

[FX-3266]: https://artsyproduct.atlassian.net/browse/FX-3266

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (Gracefully respond when Algolia API key expires - dimatretyak)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 718c5cdd93e47a127f2e0e46c82e28cb18523781
</p>

File: src/lib/Scenes/Search2/types.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export interface AlgoliaSearchResult {
+  href: string
+  image_url: string
+  name: string
+  objectID: string
+  slug: string
+  __position: number
+  __queryID: string
+}
```
Comment (gkartalis) [inline src/lib/Scenes/Search2/types.ts:9]: 🤩

**Review by gkartalis — APPROVED** [review]
Really nice solution @dimatretyak!

**Comment by dimatretyak** [comment]
> Really nice solution @dimatretyak!

Thanks 🤝

**Pull request merged**